common/djangoapps/student/models.py

```diff
@@ -1291,6 +1311,54 @@ def is_verified_enrollment(self):
         return CourseMode.is_verified_slug(self.mode)
 
 
+class ManualEnrollmentAudit(models.Model):
+    """
+    Table for track which enrollments were performed through manual enrollment.
+    """
+    enrollment = models.ForeignKey(CourseEnrollment, null=True)
+    enrolled_by = models.ForeignKey(User, null=True)
+    enrolled_email = models.CharField(max_length=255, db_index=True)
```

Comment:
@chrisndodge Once the user is entered into the AllowedEnrollments table, then after that he/she will register/activate his/her account and after account activation we have to add a new entry in the ManualEnrollmentAudit table with the state_transition to ALLOWEDTOENROLL_TO_ENROLL and at that time we don't have information `enrolled_by` and `reason` fields. So we get that information by fetching the latest manual_enrollment_audit entry with the enrolled_email field.